draft-ietf-ppm-dap.md

```diff
@@ -2662,6 +2662,10 @@ the Collector's PUT until the collection job is ready, in which case it responds
 with the `CollectionJobResp`, or the job fails, in which case the Leader MUST
 abort with the error that caused the failure.
 
+If the job fails with `invalidBatchSize`, then the Collector MAY retry it later,
+once it believes enough new reports have been uploaded and aggregated to allow
+the collection job to succeed.
```

Comment:
At first I thought this failure was impossible, but I now see that it depends on new behavior in this PR. Currently (before this PR) I believe the Leader accepts a collection job regardless of how many reports are in the batch so far and only completes the job once a sufficient number of reports have been aggregated. Now (as of this PR) the leader can quit if it doesn't thin it's going to get more reports. Is my understanding correct?

Reply:
Yeah, well spotted, that is a change worth discussing and highlighting. The previous text about this was:

```
Next, the Aggregator checks that the number of reports in the batch is equal to
or greater than the task's minimum batch size. If not, then the Helper MUST
abort with error `invalidBatchSize`. The Leader SHOULD wait for more reports to
be validated and try the collection job again later.
```

That was a SHOULD, which means the Leader was also permitted to give up on the job. We just didn't specify what error it should give up with (or indeed that it should give up at all). So I think this is something that could happen before anyway, but with this change, we're more clear about what the Collector should be ready to handle.

Reply:
Oh, and for context, the new text describing the Leader behavior:

```
If the number of validated reports in the batch is not equal to or greater than
the task's minimum batch size, then the Leader SHOULD wait for more reports to
be uploaded and aggregated and try the collection job again later. The Leader
MAY give up on the collection job (for example, if it decides that no new
reports satisfying the query are likely to ever arrive), in which case it MUST
fail the job with error `invalidBatchSize`.
```

So it's the same SHOULD, but now we offer another option.

Reply:
Oh yes, I see! This text is strictly clearer. I'm happy.